lib/kernel/src/net_kernel.erl

```diff
@@ -362,21 +366,112 @@ retry_request_maybe(Req) ->
 %% This function is used to dynamically start the
 %% distribution.
 
+-spec start(Args) -> {ok, pid()} | {error, Reason} when
+      Args :: nonempty_list(Arg),
+      Arg :: {name, Name}
+           | {name_domain, NameDomain}
+           | {net_ticktime, NetTickTime}
+           | {net_tickintensity, NetTickIntensity},
```

Comment:
I implemented a version with options in a map instead of the 2-tuple list and I liked it, so I think I lean towards using option maps instead of option lists now.